File: Benchmarks/enterprise/java/spring/testcode/BenchmarkTest58683.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import javax.xml.parsers.*;
import org.springframework.web.bind.annotation.*;

@RestController
public class BenchmarkTest58683 {

    @PostMapping(path="/BenchmarkTest58683", consumes="application/xml")
    public void BenchmarkTest58683(@RequestBody String xmlBody, HttpServletRequest request, HttpServletResponse response) throws Exception {
        String xmlValue = xmlBody;
        String data = String.format("payload=%s", xmlValue);
        DocumentBuilderFactory.newInstance().newDocumentBuilder().parse(new org.xml.sax.InputSource(new java.io.StringReader(data)));
        response.setContentType("application/json");
        response.getWriter().print("{\"id\":0}");
    }
}

File: Benchmarks/enterprise/python/fastapi/testcode/benchmark_test_58683.py
# SPDX-License-Identifier: Apache-2.0
from fastapi import Request


async def BenchmarkTest58683(request: Request):
    xml_value = (await request.body()).decode('utf-8')
    data = ' '.join(str(xml_value).split())
    request.session['data'] = str(data)
    return {"updated": True}
